core/src/main/scala/cromwell/core/retry/Retry.scala

```diff
@@ -27,7 +27,8 @@ object Retry {
                    maxRetries: Option[Int] = Option(10),
                    backoff: Backoff = SimpleExponentialBackoff(5 seconds, 10 seconds, 1.1D),
                    isTransient: Throwable => Boolean = throwableToFalse,
-                   isFatal: Throwable => Boolean = throwableToFalse)
+                   isFatal: Throwable => Boolean = throwableToFalse,
+                   onRetry: Throwable => Unit = noopOnRetry)
```

Comment:
Yep that does look close ! Since this Retry stuff is used a lot I don't want to make this PR even bigger by replacing it everywhere with fs2 but I agree it would be good to do